### ISSUE TITLE
wrong multiband factor, segfault in dwirecon loading initialisation...

### ISSUE THREAD

**Issue by maxpietsch** [dchristiaens/shard-recon#4]
`dwimotioncorrect dwirecon/premc-dwi.mif.gz out/postmc-mssh.mif.gz -fieldmap topup_fieldcoef.nii.gz -mask dwirecon/reconmask.mif.gz -pe_table dwirecon/acqp.txt -export_motion out/motion.txt -export_weights out/weights.txt -scratch /tmp -nocleanup -debug` segfaults when initialising the second recon with `recon-0.mif`.

I don't know if it's related but `Multiband factor 48 detected.` is definitely off, setting `-mb 1` in `dwirecon` does not change this message and does not solve the crash.

```
lldb ~/dwirecon/bin/dwirecon -- unwarped.mif recon-1.mif -spred spred.mif -maxiter 3 -reg 0.005 -zreg 0.001 -rf rf1.txt -rf rf2.txt -rf rf3.txt -rf rf4.txt -motion motion.txt -weights sliceweights.txt -init recon-0.mif -force -debug -nth 0
(lldb) target create "/Users/mp/dwirecon/bin/dwirecon"
Current executable set to '/Users/mp/dwirecon/bin/dwirecon' (x86_64).
(lldb) settings set -- target.run-args  "unwarped.mif" "recon-1.mif" "-spred" "spred.mif" "-maxiter" "3" "-reg" "0.005" "-zreg" "0.001" "-rf" "rf1.txt" "-rf" "rf2.txt" "-rf" "rf3.txt" "-rf" "rf4.txt" "-motion" "motion.txt" "-weights" "sliceweights.txt" "-init" "recon-0.mif" "-force" "-debug" "-nth" "0"
(lldb) run
Process 75759 launched: '/Users/mp/dwirecon/bin/dwirecon' (x86_64)
dwirecon: [WARNING] existing output files will be overwritten
dwirecon: [DEBUG] No config file found at "/etc/mrtrix.conf"
dwirecon: [INFO] reading config file "/Users/mp/.mrtrix.conf"...
dwirecon: [DEBUG] reading key/value file "/Users/mp/.mrtrix.conf"...
dwirecon: [INFO] opening image "unwarped.mif"...
dwirecon: [DEBUG] reading key/value file "unwarped.mif"...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] image "unwarped.mif" opened with dimensions 88x88x48x75, voxel spacing 2.5x2.5x2.5x6.7000000000000002, datatype Float32LE
dwirecon: [DEBUG] memory-mapping file "unwarped.mif"...
dwirecon: [DEBUG] file "unwarped.mif" mapped at 0x102000000, size 111513600 (read-only)
dwirecon: [DEBUG] image "unwarped.mif" loaded
dwirecon: [DEBUG] image "unwarped.mif" initialised with strides = [ -1 88 7744 371712 ], start = 87, using direct IO
dwirecon: [DEBUG] loading matrix file "motion.txt"...
dwirecon: [DEBUG] found 75x6 matrix in file "motion.txt"
dwirecon: [DEBUG] searching for suitable gradient encoding...
dwirecon: [DEBUG] b-value scaling: max scaling factor = exp(1.2971224095667802e-10) = 1.0000000001297122
dwirecon: [INFO] found 75x4 diffusion gradient table
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] No DW shells to be removed
dwirecon: [DEBUG] loading matrix file "rf1.txt"...
dwirecon: [DEBUG] found 4x3 matrix in file "rf1.txt"
dwirecon: [DEBUG] loading matrix file "rf2.txt"...
dwirecon: [DEBUG] found 4x3 matrix in file "rf2.txt"
dwirecon: [DEBUG] loading matrix file "rf3.txt"...
dwirecon: [DEBUG] found 4x2 matrix in file "rf3.txt"
dwirecon: [DEBUG] loading matrix file "rf4.txt"...
dwirecon: [DEBUG] found 4x1 matrix in file "rf4.txt"
dwirecon: [DEBUG] loading matrix file "sliceweights.txt"...
dwirecon: [DEBUG] found 48x75 matrix in file "sliceweights.txt"
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] initialise Y
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] Multiband factor 48 detected.
dwirecon: [INFO] initialise reconstruction matrix
dwirecon: [DEBUG] Initialising Laplacian regularizer.
dwirecon: [DEBUG] Initialising slice regularizer.
dwirecon: [100%] loading image data
dwirecon: [INFO] initialise conjugate gradient solver
dwirecon: [INFO] opening image "recon-0.mif"...
dwirecon: [DEBUG] reading key/value file "recon-0.mif"...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] image "recon-0.mif" opened with dimensions 88x88x48x4x15, voxel spacing 2.5x2.5x2.5x6.7000000000000002xnan, datatype Float32LE
dwirecon: [DEBUG] memory-mapping file "recon-0.mif"...
dwirecon: [DEBUG] file "recon-0.mif" mapped at 0x13e000000, size 89210880 (read-only)
dwirecon: [DEBUG] image "recon-0.mif" loaded
dwirecon: [DEBUG] image "recon-0.mif" initialised with strides = [ 60 5280 464640 15 1 ], start = 0, using direct IO
dwirecon: [  0%] loading initialisation...Process 75759 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
    frame #0: 0x000000010000ba5b dwirecon`run() + 24683
dwirecon`run:
->  0x10000ba5b <+24683>: movss  (%rax,%rcx,4), %xmm1      ; xmm1 = mem[0],zero,zero,zero
    0x10000ba60 <+24688>: ucomiss %xmm1, %xmm1
    0x10000ba63 <+24691>: xorps  %xmm0, %xmm0
    0x10000ba66 <+24694>: jnp    0x10000baf7               ; <+24839>
Target 0: (dwirecon) stopped.
```

mrtrix 3.0.2, 4ab54489f40997f7da1e1915c2adde3373cf6039

<details>
<summary>config</summary>
<p>

```
# Detecting OS: darwin
# Looking for compiler [clang++]: Apple clang version 12.0.0 (clang-1200.0.32.28)
# Checking for C++11 compliance: ok
# Checking shared library generation: ok
# Detecting pointer size: 64 bit
# Detecting byte order: little-endian
# Checking for variable-length array support: ok
# Checking for non-POD variable-length array support: ok
# Checking for ::max_align_t: 16 bytes
# Checking for std::max_align_t: 16 bytes
# Checking for Eigen3 library: 3.3.8
# Checking for Eigen3 Unsupported: Present
# Checking for zlib compression library: 1.2.11
# Checking for "JSON for Modern C++" requirements: ok
# Checking for TIFF library: LIBTIFF, Version 4.1.0
# Checking for PNG library: Header: 1.6.37; library: 1.6.37
# Checking for FFTW library: fftw-3.3.8-sse2-avx
# Checking for Qt moc: moc (version 5.15.1)
# Checking for Qt qmake: qmake (version 5.15.1)
# Checking for Qt rcc: rcc (version 5.15.1)
# Checking for Qt: 5.15.1


PATH = r'/Users/mp/src/ANTs/install/bin/:/Users/mp/mrtrix3_mrreg_lite/bin:/usr/local/opt/qt/bin:/usr/local/fsl/bin:/Users/mp/Applications/mrtrix3/mrtrix3/bin:/Users/mp/mrtrix3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Apple/usr/bin'
obj_suffix = '.o'
exe_suffix = ''
lib_prefix = 'lib'
lib_suffix = '.dylib'
cpp = [ 'clang++', '-c', 'CFLAGS', 'SRC', '-o', 'OBJECT' ]
cpp_flags = [ '-std=c++11', '-DMRTRIX_BUILD_TYPE="release version"', '-DMRTRIX_MACOSX', '-fPIC', '-mmacosx-version-min=11.1', '-DMRTRIX_WORD64', '-DMRTRIX_HAVE_EIGEN_UNSUPPORTED_SPECIAL_FUNCTIONS', '-DMRTRIX_TIFF_SUPPORT', '-idirafter', '/usr/local/Cellar/libtiff/4.1.0_1/include', '-DMRTRIX_PNG_SUPPORT', '-idirafter', '/usr/local/Cellar/libpng/1.6.37/include/libpng16', '-DEIGEN_FFTW_DEFAULT', '-idirafter', '/usr/local/Cellar/fftw/3.3.8_2/include', '-Wall', '-O3', '-DNDEBUG' ]
ld = [ 'clang++', 'OBJECTS', 'LINKFLAGS', '-o', 'EXECUTABLE' ]
ld_flags = [ '-mmacosx-version-min=11.1', '-lz', '-L/usr/local/Cellar/libtiff/4.1.0_1/lib', '-ltiff', '-L/usr/local/Cellar/libpng/1.6.37/lib', '-lpng16', '-lz', '-L/usr/local/Cellar/fftw/3.3.8_2/lib', '-lfftw3' ]
runpath = '-Wl,-rpath,@loader_path/'
ld_enabled = True
ld_lib = [ 'clang++', 'OBJECTS', 'LINKLIB_FLAGS', '-o', 'LIB' ]
ld_lib_flags = [ '-dynamiclib', '-install_name', '@rpath/LIBNAME', '-mmacosx-version-min=11.1', '-lz', '-L/usr/local/Cellar/libtiff/4.1.0_1/lib', '-ltiff', '-L/usr/local/Cellar/libpng/1.6.37/lib', '-lpng16', '-lz', '-L/usr/local/Cellar/fftw/3.3.8_2/lib', '-lfftw3' ]
eigen_cflags = [ '-idirafter', '/usr/local/Cellar/eigen/3.3.8_1/include/eigen3', '-DEIGEN_DONT_PARALLELIZE' ]
moc = 'moc'
rcc = 'rcc'
qt_cflags = [ '-pipe', '-stdlib=libc++', '-std=gnu++11', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk', '-mmacosx-version-min=11.1', '-Wextra', '-fPIC', '-DQT_NO_DEBUG', '-DQT_OPENGL_LIB', '-DQT_SVG_LIB', '-DQT_WIDGETS_LIB', '-DQT_GUI_LIB', '-DQT_NETWORK_LIB', '-DQT_CORE_LIB', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtOpenGL.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtSvg.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtWidgets.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtGui.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtNetwork.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/Headers', '-idirafter', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/OpenGL.framework/Headers', '-idirafter', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AGL.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/mkspecs/macx-clang', '-F/usr/local/Cellar/qt/5.15.1/lib' ]
qt_ldflags = [ '-stdlib=libc++', '-headerpad_max_install_names', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk', '-mmacosx-version-min=11.1', '-Wl,-rpath,@executable_path/../Frameworks', '-F/usr/local/Cellar/qt/5.15.1/lib', '-framework', 'QtOpenGL', '-framework', 'QtSvg', '-framework', 'QtWidgets', '-framework', 'QtGui', '-framework', 'AppKit', '-framework', 'Metal', '-framework', 'QtNetwork', '-framework', 'QtCore', '-framework', 'DiskArbitration', '-framework', 'IOKit', '-framework', 'OpenGL', '-framework', 'AGL' ]
nogui = False
```

</p>
</details>  

--- 

using the same files, I cannot reproduce it with an assert or debug build:

```
$ lldb ~/dwirecon/bin/dwirecon -- unwarped.mif recon-1.mif -spred spred.mif -maxiter 3 -reg 0.005 -zreg 0.001 -rf rf1.txt -rf rf2.txt -rf rf3.txt -rf rf4.txt -motion motion.txt -weights sliceweights.txt -init recon-0.mif -force -debug -nth 0

dwirecon: [WARNING] existing output files will be overwritten
dwirecon: [DEBUG] No config file found at "/etc/mrtrix.conf"
dwirecon: [INFO] reading config file "/Users/mp/.mrtrix.conf"...
dwirecon: [DEBUG] reading key/value file "/Users/mp/.mrtrix.conf"...
dwirecon: [INFO] opening image "unwarped.mif"...
dwirecon: [DEBUG] reading key/value file "unwarped.mif"...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] image "unwarped.mif" opened with dimensions 88x88x48x75, voxel spacing 2.5x2.5x2.5x6.7000000000000002, datatype Float32LE
dwirecon: [DEBUG] memory-mapping file "unwarped.mif"...
dwirecon: [DEBUG] file "unwarped.mif" mapped at 0x105000000, size 111513600 (read-only)
dwirecon: [DEBUG] image "unwarped.mif" loaded
dwirecon: [DEBUG] image "unwarped.mif" initialised with strides = [ -1 88 7744 371712 ], start = 87, using direct IO
dwirecon: [DEBUG] loading matrix file "motion.txt"...
dwirecon: [DEBUG] found 75x6 matrix in file "motion.txt"
dwirecon: [DEBUG] searching for suitable gradient encoding...
dwirecon: [DEBUG] b-value scaling: max scaling factor = exp(1.2971224095667802e-10) = 1.0000000001297122
dwirecon: [INFO] found 75x4 diffusion gradient table
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] No DW shells to be removed
dwirecon: [DEBUG] loading matrix file "rf1.txt"...
dwirecon: [DEBUG] found 4x3 matrix in file "rf1.txt"
dwirecon: [DEBUG] loading matrix file "rf2.txt"...
dwirecon: [DEBUG] found 4x3 matrix in file "rf2.txt"
dwirecon: [DEBUG] loading matrix file "rf3.txt"...
dwirecon: [DEBUG] found 4x2 matrix in file "rf3.txt"
dwirecon: [DEBUG] loading matrix file "rf4.txt"...
dwirecon: [DEBUG] found 4x1 matrix in file "rf4.txt"
dwirecon: [DEBUG] loading matrix file "sliceweights.txt"...
dwirecon: [DEBUG] found 48x75 matrix in file "sliceweights.txt"
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] initialise Y
dwirecon: [INFO] Diffusion gradient encoding data clustered into 3 non-zero shells and 3 b=0 volumes
dwirecon: [DEBUG] Shells: b = { 0(3) 700(12) 1500(30) 2000(30) }
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] Multiband factor 48 detected.
dwirecon: [INFO] initialise reconstruction matrix
dwirecon: [DEBUG] Initialising Laplacian regularizer.
dwirecon: [DEBUG] Initialising slice regularizer.
dwirecon: [100%] loading image data
dwirecon: [INFO] initialise conjugate gradient solver
dwirecon: [INFO] opening image "recon-0.mif"...
dwirecon: [DEBUG] reading key/value file "recon-0.mif"...
dwirecon: [DEBUG] sanitising image information...
dwirecon: [INFO] image "recon-0.mif" opened with dimensions 88x88x48x4x15, voxel spacing 2.5x2.5x2.5x6.7000000000000002xnan, datatype Float32LE
dwirecon: [DEBUG] memory-mapping file "recon-0.mif"...
dwirecon: [DEBUG] file "recon-0.mif" mapped at 0x13a016000, size 89210880 (read-only)
dwirecon: [DEBUG] image "recon-0.mif" loaded
dwirecon: [DEBUG] image "recon-0.mif" initialised with strides = [ 60 5280 464640 15 1 ], start = 0, using direct IO
dwirecon: [100%] loading initialisation
dwirecon: [INFO] solve from given starting point
dwirecon: [INFO] Forward projection.
dwirecon: [DEBUG] image view "unwarped.mif" initialised with strides = [ 37 3256 286528 1 ], start = 0
dwirecon: [DEBUG] image view "unwarped.mif" initialised with strides = [ 1 88 7744 371712 ], start = 0
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] allocating scratch buffer for image "temporary buffer"...
dwirecon: [DEBUG] image "temporary buffer" loaded
dwirecon: [DEBUG] image "temporary buffer" initialised with strides = [ 1 88 7744 ], start = 0, using direct IO
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] allocating scratch buffer for image "temporary buffer"...
dwirecon: [DEBUG] image "temporary buffer" loaded
dwirecon: [DEBUG] image "temporary buffer" initialised with strides = [ 1 88 7744 ], start = 0, using direct IO
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] allocating scratch buffer for image "temporary buffer"...
dwirecon: [DEBUG] image "temporary buffer" loaded
dwirecon: [DEBUG] image "temporary buffer" initialised with strides = [ 1 88 7744 ], start = 0, using direct IO
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] allocating scratch buffer for image "temporary buffer"...
dwirecon: [DEBUG] image "temporary buffer" loaded
dwirecon: [DEBUG] image "temporary buffer" initialised with strides = [ 1 88 7744 ], start = 0, using direct IO
dwirecon: [DEBUG] sanitising image information...
dwirecon: [DEBUG] allocating scratch buffer for image "temporary buffer"...
dwirecon: [DEBUG] image "temporary buffer" loaded
dwirecon: [DEBUG] image "temporary buffer" initialised with strides = [ 1 88 7744 ], start = 0, using direct IO
dwirecon: [  4%] forward projection...

```

<details>
<summary>config</summary>
<p>

```
# Detecting OS: darwin
# Looking for compiler [clang++]: Apple clang version 12.0.0 (clang-1200.0.32.28)
# Checking for C++11 compliance: ok
# Checking shared library generation: ok
# Detecting pointer size: 64 bit
# Detecting byte order: little-endian
# Checking for variable-length array support: ok
# Checking for non-POD variable-length array support: ok
# Checking for ::max_align_t: 16 bytes
# Checking for std::max_align_t: 16 bytes
# Checking for Eigen3 library: 3.3.8
# Checking for Eigen3 Unsupported: Present
# Checking for zlib compression library: 1.2.11
# Checking for "JSON for Modern C++" requirements: ok
# Checking for TIFF library: LIBTIFF, Version 4.1.0
# Checking for PNG library: Header: 1.6.37; library: 1.6.37
# Checking for FFTW library: fftw-3.3.8-sse2-avx
# Checking for Qt moc: moc (version 5.15.1)
# Checking for Qt qmake: qmake (version 5.15.1)
# Checking for Qt rcc: rcc (version 5.15.1)
# Checking for Qt: 5.15.1


PATH = r'/Users/mp/src/ANTs/install/bin/:/Users/mp/mrtrix3_mrreg_lite/bin:/usr/local/opt/qt/bin:/usr/local/fsl/bin:/Users/mp/Applications/mrtrix3/mrtrix3/bin:/Users/mp/mrtrix3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Apple/usr/bin'
obj_suffix = '.o'
exe_suffix = ''
lib_prefix = 'lib'
lib_suffix = '.dylib'
cpp = [ 'clang++', '-c', 'CFLAGS', 'SRC', '-o', 'OBJECT' ]
cpp_flags = [ '-std=c++11', '-DMRTRIX_BUILD_TYPE="debug version with nooptim"', '-DMRTRIX_MACOSX', '-fPIC', '-mmacosx-version-min=11.1', '-DMRTRIX_WORD64', '-DMRTRIX_HAVE_EIGEN_UNSUPPORTED_SPECIAL_FUNCTIONS', '-DMRTRIX_TIFF_SUPPORT', '-idirafter', '/usr/local/Cellar/libtiff/4.1.0_1/include', '-DMRTRIX_PNG_SUPPORT', '-idirafter', '/usr/local/Cellar/libpng/1.6.37/include/libpng16', '-DEIGEN_FFTW_DEFAULT', '-idirafter', '/usr/local/Cellar/fftw/3.3.8_2/include', '-Wall', '-g', '-O0' ]
ld = [ 'clang++', 'OBJECTS', 'LINKFLAGS', '-o', 'EXECUTABLE' ]
ld_flags = [ '-mmacosx-version-min=11.1', '-lz', '-L/usr/local/Cellar/libtiff/4.1.0_1/lib', '-ltiff', '-L/usr/local/Cellar/libpng/1.6.37/lib', '-lpng16', '-lz', '-L/usr/local/Cellar/fftw/3.3.8_2/lib', '-lfftw3', '-g' ]
runpath = '-Wl,-rpath,@loader_path/'
ld_enabled = True
ld_lib = [ 'clang++', 'OBJECTS', 'LINKLIB_FLAGS', '-o', 'LIB' ]
ld_lib_flags = [ '-dynamiclib', '-install_name', '@rpath/LIBNAME', '-mmacosx-version-min=11.1', '-lz', '-L/usr/local/Cellar/libtiff/4.1.0_1/lib', '-ltiff', '-L/usr/local/Cellar/libpng/1.6.37/lib', '-lpng16', '-lz', '-L/usr/local/Cellar/fftw/3.3.8_2/lib', '-lfftw3', '-g' ]
eigen_cflags = [ '-idirafter', '/usr/local/Cellar/eigen/3.3.8_1/include/eigen3', '-DEIGEN_DONT_PARALLELIZE' ]
moc = 'moc'
rcc = 'rcc'
qt_cflags = [ '-pipe', '-stdlib=libc++', '-g', '-std=gnu++11', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk', '-mmacosx-version-min=11.1', '-Wextra', '-fPIC', '-DQT_OPENGL_LIB', '-DQT_SVG_LIB', '-DQT_WIDGETS_LIB', '-DQT_GUI_LIB', '-DQT_NETWORK_LIB', '-DQT_CORE_LIB', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtOpenGL.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtSvg.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtWidgets.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtGui.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtNetwork.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/Headers', '-idirafter', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/OpenGL.framework/Headers', '-idirafter', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AGL.framework/Headers', '-idirafter', '/usr/local/Cellar/qt/5.15.1/mkspecs/macx-clang', '-F/usr/local/Cellar/qt/5.15.1/lib' ]
qt_ldflags = [ '-stdlib=libc++', '-headerpad_max_install_names', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk', '-mmacosx-version-min=11.1', '-Wl,-rpath,@executable_path/../Frameworks', '-F/usr/local/Cellar/qt/5.15.1/lib', '-framework', 'QtOpenGL', '-framework', 'QtSvg', '-framework', 'QtWidgets', '-framework', 'QtGui', '-framework', 'AppKit', '-framework', 'Metal', '-framework', 'QtNetwork', '-framework', 'QtCore', '-framework', 'DiskArbitration', '-framework', 'IOKit', '-framework', 'OpenGL', '-framework', 'AGL' ]
nogui = False
```

</p>
</details>  

I tried all of this twice, starting with `dwimotioncorrect`, (once mrtrix 3.0.2 (4ab54489f40997f7da1e1915c2adde3373cf6039) and once with the latest master 51f0fc69d2bbb469403a6656ff9f1e5817425d1a).

**Comment by dchristiaens** [comment]
`Multiband factor 48 detected.` just means that `dwirecon` will run volume-level instead of slice-level reconstruction, as is intended in the first iterations. MB=48 then means that all 48 slices are treated as if they are acquired simultaneously. The MB factor is not set in `dwirecon`, but inferred from the data size and the number of entries in the motion file.

Not sure where that segfault originates from. Does `recon-0.mif` look OK?

**Comment by maxpietsch** [comment]
`recon-0.mif` and `unwarped.mif` look ok. I tried two different subjects/sessions and both segfault. 

It crashes somewhere [here](https://github.com/dchristiaens/shard-recon/blob/master/cmd/dwirecon.cpp#L348-L350), I think because `k` exceeds `c.rows()` (it does most of the time I run the command). In case it's needed, I sent you the input data and temporary folder content.

**Comment by dchristiaens** [comment]
OK, a small refactoring of the code seems to fix it (#8), although I'm not quite sure why there is a difference.
